services/notifications/service/notifications_service.go

```diff
@@ -355,6 +355,60 @@ func RegisterDeviceToUser(token string, platform string, id string) error {
 	return nil
 }
 
+/*
+	Unregisters the device token with SNS and removes the arn from the associated user
+*/
+func UnregisterDeviceFromUser(token string, platform string, id string) error {
+	var platform_arn string
+
+	switch strings.ToLower(platform) {
+	case "android":
+		platform_arn = config.ANDROID_PLATFORM_ARN
+	case "ios":
+		platform_arn = config.IOS_PLATFORM_ARN
+	default:
+		return errors.New("Invalid platform")
+	}
+
+	var device_arn string
+
+	if config.IS_PRODUCTION {
+		response, err := client.CreatePlatformEndpoint(
+			&sns.CreatePlatformEndpointInput{
+				CustomUserData:         &id,
+				Token:                  &token,
+				PlatformApplicationArn: &platform_arn,
+			},
+		)
+
+		if err != nil {
+			return err
+		}
+
+		device_arn = *response.EndpointArn
+	}
+
+	devices, err := GetUserDevices(id)
+
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(devices); i++ {
+		if devices[i] == device_arn {
+			devices = append(devices[:i], devices[i+1:]...)
```

Comment:
It would be good if we could write a test for this functionality. Since tests have `IS_PRODUCTION` false, we should be able to test the db functionality fine.

Reply:
Definitely. Do we just write any tests for our program in the /gateway/tests/gateway_test.go or is there another place where we write them?

Reply:
Tests go in the specific `/services/notifications/service/tests/` folder